aptos-move/move-examples/token_objects/knight/sources/knight.move

```diff
@@ -0,0 +1,541 @@
+
+        // Initializes the value with the given value in food.
+        move_to(&object_signer, RestorationValue { value: restoration_value });
+
+        // Initialize the property map.
+        let properties = property_map::prepare_input(vector[], vector[], vector[]);
+        property_map::init(&constructor_ref, properties);
+        property_map::add_typed(
+            &property_mutator_ref,
+            string::utf8(b"Restoration Value"),
+            string_utils::to_string(&restoration_value)
+        );
+
+        // Creates the fungible asset.
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            &constructor_ref,
+            option::none(),
+            fungible_asset_name,
+            fungible_asset_symbol,
+            0,
+            icon_uri,
+            project_uri,
+        );
+        let fungible_asset_mint_ref = fungible_asset::generate_mint_ref(&constructor_ref);
+        let fungible_asset_burn_ref = fungible_asset::generate_burn_ref(&constructor_ref);
+        let fungible_asset_transfer_ref = fungible_asset::generate_transfer_ref(&constructor_ref);
+
+        // Publishes the FoodToken resource with the refs.
+        let food_token = FoodToken {
+            burn_ref,
+            property_mutator_ref,
+            fungible_asset_mint_ref,
+            fungible_asset_transfer_ref,
+            fungible_asset_burn_ref,
+        };
+        move_to(&object_signer, food_token);
+    }
+
+    /// Mints an knight token. This function mints a new knight token and transfers it to the
+    /// `soul_bound_to` address. The token is minted with health point 0 and condition Hungry.
+    public entry fun mint_knight(
+        creator: &signer,
+        description: String,
+        name: String,
+        base_uri: String,
+        receiver: address,
+    ) {
+        // The collection name is used to locate the collection object and to create a new token object.
+        let collection = string::utf8(KNIGHT_COLLECTION_NAME);
+        // Creates the knight token, and get the constructor ref of the token. The constructor ref
+        // is used to generate the refs of the token.
+        let uri = base_uri;
+        string::append(&mut uri, string::utf8(CONDITION_HUNGRY));
+        let constructor_ref = token::create_named_token(
+            creator,
+            collection,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+
+        // Generates the object signer and the refs. The object signer is used to publish a resource
+        // (e.g., knightLevel) under the token object address. The refs are used to manage the token.
+        let object_signer = object::generate_signer(&constructor_ref);
+        let transfer_ref = object::generate_transfer_ref(&constructor_ref);
+        let mutator_ref = token::generate_mutator_ref(&constructor_ref);
+        let burn_ref = token::generate_burn_ref(&constructor_ref);
```

Comment:
Do we actually use the burn ref for knights in this example? Let's remove all refs that we don't need just for simplicity

Reply:
Done.